node/src/neighborhood/mod.rs

```diff
@@ -1835,11 +1840,58 @@ mod tests {
         System::current().stop();
         assert_eq!(system.run(), 0);
         TestLogHandler::new().exists_log_containing(&format!(
-            "TRACE: Neighborhood: An unnecessary ConnectionProgressMessage received from IP Address: {:?}",
+            "TRACE: Neighborhood: An unnecessary ConnectionProgressMessage received containing IP Address: {:?}",
             unknown_peer
         ));
     }
 
+    #[test]
+    fn neighborhood_logs_with_trace_if_it_receives_a_cpm_with_a_pass_target_that_is_a_part_of_a_different_connection_progress(
+    ) {
+        init_test_logging();
+        let peer_1 = make_ip(1);
+        let peer_2 = make_ip(2);
+        let this_node_addr = NodeAddr::new(&IpAddr::from_str("111.111.111.111").unwrap(), &[8765]);
+        let initial_node_descriptors =
+            vec![make_node_descriptor(peer_1), make_node_descriptor(peer_2)];
+        let neighborhood_config = NeighborhoodConfig {
+            mode: NeighborhoodMode::Standard(
+                this_node_addr,
+                initial_node_descriptors,
+                rate_pack(100),
+            ),
+        };
+        let bootstrap_config =
+            bc_from_nc_plus(neighborhood_config, make_wallet("earning"), None, "test");
+        let mut subject = Neighborhood::new(main_cryptde(), &bootstrap_config);
+        subject
+            .overall_connection_status
+            .get_connection_progress_by_ip(peer_1)
+            .unwrap()
+            .connection_stage = ConnectionStage::TcpConnectionEstablished;
+        subject
+            .overall_connection_status
+            .get_connection_progress_by_ip(peer_2)
+            .unwrap()
+            .connection_stage = ConnectionStage::TcpConnectionEstablished;
+        let addr = subject.start();
+        let cpm_recipient = addr.clone().recipient::<ConnectionProgressMessage>();
+        let system = System::new("testing");
+        let cpm = ConnectionProgressMessage {
+            peer_addr: peer_2,
+            event: ConnectionProgressEvent::PassGossipReceived(peer_1),
+        };
+
+        cpm_recipient.try_send(cpm).unwrap();
+
+        System::current().stop();
+        assert_eq!(system.run(), 0);
+        TestLogHandler::new().exists_log_containing(&format!(
+            "TRACE: Neighborhood: We've been passed to a peer with IP Address: {:?} that's already a part of different connection progress.",
```

Comment:
This message implies that the pass actually happened. Can you reword so that it's clear that the pass was attempted but rejected?

Reply:
I have reworded it.